ecommerce/models.py

```diff
@@ -1,3 +1,59 @@
 from django.db import models
+from django.urls import reverse
 
-# Create your models here.
+class Category(models.Model):
+    name = models.CharField(max_length=200, help_text='Enter a product category (e.g. Fashion Toy)')
+
+    def __str__(self):
+        return self.name
+
+class Product(models.Model):
+    product_name = models.CharField(max_length=200)
+    image = models.ImageField(default='/images/product_pics/default.png', upload_to='images/product_pics')
+    category = models.ForeignKey('Category', on_delete=models.SET_NULL, null=True)
+    description = models. TextField(max_length=1000, help_text='Enter a brief description of the product')
+    publish_date = models.DateField(null=True, blank=True)
+    price = models.FloatField(blank=True, null=True)
+    quantity = models.IntegerField(null=True)
+
+    def __str__(self):
+        return self.product_name
+
+    def get_absolute_url(self):
+        return reverse('product-detail', args=[str(self.id)])
+
+class User(models.Model):
+    username = models.CharField(max_length=200)
+    email = models.EmailField(max_length=200)
+    image = models.ImageField(default='/images/profile_pics/default.jpg', upload_to='images/profile_pics')
+    birthday = models.DateField(null=True,blank=True)
+    address = models.CharField(max_length=200)
+    
+    def __str__(self):
+        return self.username
+
+class Order(models.Model):
+    user = models.ForeignKey('User', on_delete=models.CASCADE, null=True)
+    product = models.ForeignKey('Product', on_delete=models.CASCADE, null=True)
+    total_price = models.FloatField(blank=True, null= True)
+    date = models.DateField(null=True, blank=True)
+
+    def __str__(self):
+        return self.user.username
+
+class FavoriteProduct(models.Model):
+    user = models.ForeignKey('User', on_delete=models.CASCADE, null=True)
+    product = models.ForeignKey('Product', on_delete=models.CASCADE, null=True)
+    favorited = models.BooleanField(default=False)
```

Comment:
đối với các column kiểu boolean thì nên đặt thêm tiền tố is_...
```suggestion
    is_favorited = models.BooleanField(default=False)
```